lib/active_merchant/billing/gateways/paymill.rb

```diff
@@ -74,11 +74,15 @@ def verify_credentials
 
       private
 
-      def add_credit_card(post, credit_card)
+      def add_credit_card(post, credit_card, money, options)
+        post['account.holder'] = (credit_card.try(:name) || "")
         post['account.number'] = credit_card.number
         post['account.expiry.month'] = sprintf("%.2i", credit_card.month)
         post['account.expiry.year'] = sprintf("%.4i", credit_card.year)
         post['account.verification'] = credit_card.verification_value
+        post['account.email'] = (credit_card.try(:email) || nil)
+        post['presentation.amount3D'] = (amount(money  * 100))
```

Comment:
I'm not sure about the `* 100` here. Since this gateway's `money_format` is already `:cents`, `#amount` will be in integer cents already.